compiler/p/codegen/FPTreeEvaluator.cpp

```diff
@@ -652,7 +652,7 @@ TR::Register *OMR::Power::TreeEvaluator::vsplatsEvaluator(TR::Node *node, TR::Co
    else if (node->getDataType() == TR::VectorInt64)
       {
       TR::Register *srcReg = cg->evaluate(child);
-      TR::Register *trgReg = cg->allocateRegister(TR_VSX_VECTOR);
+      TR::Register *trgReg = cg->allocateRegister(TR_VRF);
 
       if (!disableDirectMove && TR::Compiler->target.cpu.id() >= TR_PPCp8 && TR::Compiler->target.cpu.getPPCSupportsVSX())
```

Comment:
it looked like evaluators emitting VMX operations didn't try to/check  (but ASSERT/assuming) move the source vectors (if VSX_VECTOR not in range of VRF) to VRF. It is functionally correct to use the VRF subset of VSX_VECTOR, but you give up on half of the vector registers.